### PR TITLE
Add new task types to the task processing API

### PR DESCRIPTION
**Pull request by julien-nc**
These will be useful for the new Agency feature.

### CODE REVIEW

File: developer_manual/digging_deeper/task_processing.rst [hunk 1]
```diff
@@ -38,6 +38,24 @@ The following built-in task types are available:
        * ``history``: ``ListOfTexts``
     * Output shape:
        * ``output``: ``Text``
+ * ``'core:text2text:chatwithtools'``: This task allows chatting with the language model with tools calling support. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToTextChatWithTools``
+    * Input shape:
+       * ``system_prompt``: ``Text``
+       * ``input``: ``Text``
+       * ``history``: ``ListOfTexts``
+       * ``tools``: ``Text`` This should be a JSON object formatted like the one expected by the OpenAI API: https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools
```
Comment (oleksandr-nc) [inline developer_manual/digging_deeper/task_processing.rst:46]: The tools parameter should be a JSON object formatted according to the OpenAI API specifications: https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools

File: developer_manual/digging_deeper/task_processing.rst [hunk 2]
```diff
@@ -38,6 +38,24 @@ The following built-in task types are available:
...
+    * Output shape:
+       * ``output``: ``Text``
+       * ``tool_calls``: ``Text``
+ * ``'core:contextagent:interaction'``: This task allows chatting with an agent. It is implemented by ``\OCP\TaskProcessing\TaskTypes\ContextAgentInteraction``
+    * Input shape:
+       * ``input``: ``Text``
+       * ``confirmation``: ``Number`` Whether to confirm previously requested actions. 0 to reject, 1 to confirm.
```
Comment (oleksandr-nc) [inline developer_manual/digging_deeper/task_processing.rst:53]: `0` or **0** , `1` or **1**

or rephrased completly:

Boolean integer indicating whether to confirm previously requested actions: **0** to reject or **1** to confirm.

**Review by oleksandr-nc — APPROVED** [review]
little things, so as not to be a bore I won't describe them further =)

**Review by kyteinsky — APPROVED** [review]

File: developer_manual/digging_deeper/task_processing.rst (hunk 1 above)
Comment (julien-nc) [reply developer_manual/digging_deeper/task_processing.rst]: :+1: done

File: developer_manual/digging_deeper/task_processing.rst (hunk 2 above)
Comment (julien-nc) [reply developer_manual/digging_deeper/task_processing.rst]: :+1: done

**Pull request merged**